report/Sprint_1/Backend.tex

```diff
@@ -0,0 +1,125 @@
+    \item App update
+    \item App information retrieval
+\end{itemize}
+
+The different components of the API where implemented with their own endpoints making the calls to the API as user friendly as possible.
+
+From the list above four endpoints where created;
+\begin{itemize}
+    \item User
+    \item Task
+    \item Screen
+    \item App
+\end{itemize}
+
+For the implementation of the API the leverage points given in~\autoref{subsec:ConclusionOfLeverage} showed which technologies to utilize.
+For the API the only one of the technologies that would make sense to use is Dotnet.
+
+Dotnet is a framework developed by Microsoft using their C\# language.
+Dotnet is easy to use, in terms of development setup and production hosting and is thus a powerful tool\cite{DotnetWebsite}.
+
+The development of the API was conducted simultaneously with the development of the application it self.
+The endpoint was created as they where needed within the application.
+Allowing for the developers to work simultaneously on their issues and providing a working application along the they.
+
+The design of the API is based on the Model, View, Controller (MVC) pattern\cite{MVC}.
+However since the API does not provide any meaningful interface to view, that element is meaningless and is thus not implemented.
+
+The models are all the different objects the API has to deal with, such as users, apps, tasks and screens.
+Each model has it own class in the C\# language as can be seen in~\autoref{APIAppModel}.
+
+\begin{lstlisting}[caption={API App model}, label={APIAppModel}, language={CSharp}]
+namespace API.Models.App
+{
+    public class AppModel
+    {
+        public int Id { get; set; }
+        public string AppName { get; set; }
+        public string AppUrl { get; set; }
+        public string AppColor { get; set; }
+    }
+}
+\end{lstlisting}
+
+Each endpoint has its own controller which is used to interface with the business logic of the API.
+An example of this can be seen in~\autoref{APIAppController}.
+This snippet shows an endpoint where if the API is called on App this an HttpGet request, this function will be called and the answer returned.
+The function it self i rather simple, it calls the GetAll function and maps the result to a meaningful result that is then returned.
+
+\begin{lstlisting}[caption={API App Controller GetAll Function}, label={APIAppController}, language={CSharp}]
+/// <summary>
+/// Get All Apps in the system
+/// </summary>
+/// <returns>HttpCode 200 with a list of apps</returns>
+[HttpGet]
+public IActionResult GetAll()
+{
+    var app = _appService.GetAll();
+    var model = _mapper.Map<IList<AppModel>>(app);
+    return Ok(model);
+}
+\end{lstlisting}
+
+The actual function dealing with the logic of such a call can be seen on~\autoref{APIAppService}.
+Since this function is rather simple overall the only actual logic is accomplished by returning all apps in the datacontext, which is just a call to the database.
```

Comment:
Hvis det er BI
```suggestion
Since this function is rather simple, the actual business logic is accomplished by returning all apps in the data context, which is just a call to the database.
```

Hvis det ikke er BI
```suggestion
Since this function is rather simple, the actual logic is accomplished by returning all apps in the data context, which is just a call to the database.
```